pkgs/by-name/zm/zmk-studio/package.nix

```diff
@@ -0,0 +1,95 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  fetchNpmDeps,
+
+  cargo-tauri,
+  makeBinaryWrapper,
+  nodejs,
+  npmHooks,
+  pkg-config,
+  wrapGAppsHook3,
+
+  glib-networking,
+  udev,
+  webkitgtk_4_1,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "zmk-studio";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "zmkfirmware";
+    repo = "zmk-studio";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-g7JqALJLB9hT35ZLGKp7SAH+nj8BhJgTattEwSh8Ss8=";
+  };
+
+  patches = [
+    # generated using `npx npm-package-lock-add-resolved`
+    ./add-missing-lockfile-integrity.patch
```

Comment:
Instead of this, we can add a postFetch that uses [npm-lockfile-fix](https://search.nixos.org/packages?channel=25.05&show=npm-lockfile-fix&from=0&size=50&sort=relevance&type=packages&query=npm-lockfile-fix)

Here's an example: https://github.com/NixOS/nixpkgs/blob/nixos-25.05/pkgs/by-name/ma/mattermost/package.nix#L117C20-L117C36

Note that we also do other fixups because Mattermost's lockfile is extra broken, so you don't need those, but a single invocation of npm-lockfile-fix in a postFetch should remove the need for this patch. :-)